### PR TITLE
Back tuning issue

### PR DESCRIPTION
**Pull request by AndrePu**
1) Mapping models was taken out in separate class (Single Responsibility principe).
2) Added code for validation user properties. Can be used in registration, authentification, password changing, email/nickname validation.
3) Taken of password property in  final User model.

### CODE REVIEW

**Comment by AndrePu** [comment]
@NikBorovinsS , please, trace the changes. Ask any questions if will be a need)

File: Cooper/Controllers/RegistrationController.cs [hunk 1]
```diff
@@ -26,7 +25,12 @@ public RegistrationController()
         [ProducesResponseType(201)]
         public IActionResult Post([FromBody]User user)
         {
-            if (!ModelState.IsValid || user.Id != 0)
+            bool nicknameExists = userRepository.IfNicknameExists(user.Nickname);       // validation that we don't create user with the same nickname
+            bool emailExists = userRepository.IfEmailExists(user.Email);
+
+            // TODO: divide this statement into three and send the proper explanation for bad-request
+
+            if (!ModelState.IsValid || user.Id != 0 || nicknameExists == true || emailExists == true)
```
Comment (Hunroll) [inline Cooper/Controllers/RegistrationController.cs]: Codestyle: "nicknameExists == true" should be replaced with "nicknameExists", ==true is alway ommitted.

File: Cooper/DAO/UserDAO.cs [hunk 2]
```diff
@@ -46,6 +46,51 @@ public UserDAO()
             };
         }
 
+        #region Checking methods
+
+        public bool IfNicknameExists(string nickname)
```
Comment (Hunroll) [inline Cooper/DAO/UserDAO.cs]: Codestyle: IfNicknameExists -> NickNameExists.
Boolean properties are named IsValid, IsDisposed, .... or if it has a verb (exists, has, ...) - "is" is not needed.
So later the code will be readable : if (NicknameExists) {....}, not if (IfNicknameExists)

File: Cooper/DAO/UserDAO.cs [hunk 3]
```diff
@@ -46,6 +46,51 @@ public UserDAO()
...
+        }
+
+        public bool IfUserExists(long id)
+        {
+            UserDb user = Get(id);
+
+            return (user == null) ? false : true;
+        }
+
+        public bool CheckCredentials(string nickname, string password)
```
Comment (Hunroll) [inline Cooper/DAO/UserDAO.cs]: CodeStyle) Method name should contain its purpose. like  AreCredentialsValid

**Pull request merged**